### PR TITLE
Autocorrelation and SDRP

### PR DESCRIPTION
**Pull request by pastewka**
- BUG: Filtered wrong end of autocorrelation
- ENH: Apply reliability cutoff to scale-dependent statistical analysis

### CODE REVIEW

File: SurfaceTopography/Generic/ScaleDependentStatistics.py [hunk 1]
```diff
@@ -67,6 +67,8 @@ def scale_dependent_statistical_property(self, func, n=1, scale_factor=None, dis
         Characteristic distances at which the derivatives are computed. If
```
Comment (AntoineSIMTEK) [inline SurfaceTopography/Generic/ScaleDependentStatistics.py:67]: Could you clarify what the characteristic distance is ? If I understand the code correctly, it is scale_factor * pixel_size

File: SurfaceTopography/Generic/ScaleDependentStatistics.py [hunk 2]
```diff
@@ -121,19 +123,29 @@ def scale_dependent_statistical_property(self, func, n=1, scale_factor=None, dis
...
                 iter(scale_factor)
             if distance is not None:
                 iter(distance)
-            return distance, [func(dx, dy) for dx, dy in zip(*d)]
+            retvals = np.array([func(dx, dy) for dx, dy in zip(*d)])
         except TypeError:
+            # If there is a single distance, the reliability analysis is skipped
             return distance, func(*d)
 
+    short_cutoff = self.short_reliability_cutoff() if reliable else None
```
Comment (AntoineSIMTEK) [inline SurfaceTopography/Generic/ScaleDependentStatistics.py:145]: similar remark: I think the function short_reliability_cutoff does not clearly state what the distance scale is. To my understanding it is ell = 2 eta dx

File: SurfaceTopography/Generic/ScaleDependentStatistics.py (hunk 1 above)
Comment (pastewka) [reply SurfaceTopography/Generic/ScaleDependentStatistics.py]: I've updated the docstring. Is this clearer?

File: SurfaceTopography/Generic/ScaleDependentStatistics.py [hunk 3]
```diff
@@ -121,19 +127,29 @@ def scale_dependent_statistical_property(self, func, n=1, scale_factor=None, dis
...
+    if short_cutoff is not None:
+        mask = distance > short_cutoff / 2
```
Comment (AntoineSIMTEK) [inline SurfaceTopography/Generic/ScaleDependentStatistics.py:147]: Thanks for clarifying the docstring. But then I am a bit doubtful about this. 
For curvatures, I would cutoff data with stencil lengths smaller then the stencil length that we used for the peak curvature analysis. 

For other orders of derivatives I would make the cutoff based on the pixel size, intuitively. Any thoughts ?

File: test/test_scale_dependent_statistics.py [hunk 4]
```diff
@@ -53,7 +54,27 @@ def test_scalar_input():
...
+    f = scipy.interpolate.interp1d(r, np.sqrt(2 * A) / r)
+    r, s = t.scale_dependent_statistical_property(lambda x: np.mean(x * x), n=1)
+
+    assert abs(r.min()/L - 1) < 0.1  # With 10% of bandwidth
+    assert abs(r.max()/u - 1) < 0.1
+
+    np.testing.assert_allclose(np.sqrt(s), f(r), atol=0.01)
+
+
+def test_nonuniform_synthetic(plot=True):
```
Comment (AntoineSIMTEK) [inline test/test_scale_dependent_statistics.py:77]: should we deactivat the plot, or is that overriden by a fixture ?

**Review by AntoineSIMTEK — CHANGES_REQUESTED** [review]

File: SurfaceTopography/Generic/ScaleDependentStatistics.py (hunk 2 above)
Comment (pastewka) [reply SurfaceTopography/Generic/ScaleDependentStatistics.py:145]: This is correct, ell = 2 eta dx. I've updated the doc string to explain this.

File: test/test_scale_dependent_statistics.py (hunk 4 above)
Comment (pastewka) [reply test/test_scale_dependent_statistics.py]: There was no plotting, I've removed the argument

File: SurfaceTopography/Generic/ScaleDependentStatistics.py (hunk 3 above)
Comment (pastewka) [reply SurfaceTopography/Generic/ScaleDependentStatistics.py]: You are right. This was taken verbatim from `autocorrelation` where this criterion is correct.

**Review by AntoineSIMTEK — APPROVED** [review]

**Pull request merged**